### DISCUSSION TITLE
code errors in .bat file but not cmd

### DISCUSSION THREAD

**Discussion by kcin2001** [ImageMagick/ImageMagick#7758] [Help] [answered]
### ImageMagick version

7.1.1-39

### Operating system, version and so on

windows 10

### Description

i have a line of code that works in cmdline but not in a bat file
working cmdline
> for /r %i in (convert\fire\*.png) do magick %i -region 80x24+40+176 -fill red -colorize 50% %i

non working .bat
> for %%i in (convert\fire\*.png) do magick %%i -region 80x24+40+176 -fill red -colorize 50% %%i 

and this is the error i get

> magick: MissingArgument \`-colorize' at CLI arg 6 @ fatal/magick-cli.c/ProcessCommandOptions/676.
> magick: MissingArgument \`-colorize' at CLI arg 6 @ fatal/magick-cli.c/ProcessCommandOptions/676.

**Comment by fmw42** [comment, marked as answer]
I believe that you need to double the % on the 50% to 50%%

**Reply by kcin2001** [reply]
thank you for the quick help
